src/api/c/convolve.cpp

```diff
@@ -254,3 +300,110 @@ af_err af_convolve2_sep(af_array *out, const af_array signal, const af_array col
             return convolve2_sep<false>(out, signal, col_filter, row_filter);
     } CATCHALL;
 }
+
+template<typename T, typename accT>
+af_array conv2GradCall(const af_array incoming_gradient,
+                       const af_array original_signal,
+                       const af_array original_filter,
+                       const af_array convolved_output,
+                       af::dim4 stride, af::dim4 padding, af::dim4 dilation,
+                       af_conv_gradient_type gradType)
+{
```

Comment:
Since conv2GradCall doesn't handle complex inputs, we can remove `accT` template paramter and do the following inside this function.

```c++
using accT = typename std::conditional<std::is_same<T, double>::value, double, float>::type;
```

Reply:
This way, the switch case surrounding the `conv2GradCall`'s inside `af_convolve2Gradient` can be further trimmed down.

Actually, this removal of extra template parameter can trickle down all the way down to kernel implementation for `conv2FilterGradient`.